examples/state_dispatcher/main.rs

```diff
@@ -0,0 +1,49 @@
+//! An example showing how to create a dispatcher inside of a State.
+
+extern crate amethyst;
+
+use amethyst::ecs::{Dispatcher, DispatcherBuilder};
+use amethyst::prelude::*;
+use amethyst::Error;
+
+struct StateA;
+
+impl SimpleState<'static, 'static> for StateA {
+    fn update(&mut self, data: &mut StateData<GameData>) -> SimpleTrans<'static, 'static> {
+        println!("StateA::update()");
+        // Shows how to push a `Trans` through the event queue.
+        // If you do use TransQueue, you will be forced to use the 'static lifetime on your states.
```

Comment:
I'm curious, maybe if there is a `T: 'a` bound in `CoreApplication`, it might relax this constraint; not sure about the second lifetime though,

Reply:
```rs
error[E0310]: the parameter type `T` may not live long enough                                                                                              
   --> src/app.rs:307:40                                                                                                                                   
    |                                                                                                                                                      
137 | impl<'a, T, E, R> CoreApplication<'a, T, E, R>                                                                                                       
    |          - help: consider adding an explicit lifetime bound `T: 'static`...                                                                          
...                                                                                                                                                        
307 |                 let mut v = self.world.write_resource::<TransQueue<T, E>>();                                                                         
    |                                        ^^^^^^^^^^^^^^                                                                                                
    |                                                                                                                                                      
note: ...so that the type `std::collections::VecDeque<std::boxed::Box<(dyn std::ops::Fn() -> state::Trans<T, E> + std::marker::Send + std::marker::Sync + 'static)>>` will meet its required lifetime bounds
   --> src/app.rs:307:40                                                                                                                                   
    |                                                                                                                                                      
307 |                 let mut v = self.world.write_resource::<TransQueue<T, E>>();                                                                         
    |                                        ^^^^^^^^^^^^^^                                                                                                
                                                                                                                                                           
error[E0310]: the parameter type `T` may not live long enough                                                                                              
   --> src/app.rs:496:15                                                                                                                                   
    |                                                                                                                                                      
395 | impl<'a, S, T, E, X> ApplicationBuilder<S, T, E, X>                                                                                                  
    |             - help: consider adding an explicit lifetime bound `T: 'static`...                                                                       
...                                                                                                                                                        
496 |         world.add_resource(TransQueue::<T, StateEvent>::with_capacity(2));                                                                           
    |               ^^^^^^^^^^^^                                                                                                                           
    |                                                                                                                                                      
note: ...so that the type `std::collections::VecDeque<std::boxed::Box<dyn std::ops::Fn() -> state::Trans<T, state_event::StateEvent> + std::marker::Send + std::marker::Sync>>` will meet its required lifetime bounds
   --> src/app.rs:496:15                                                                                                                                   
    |                                                                                                                                                      
496 |         world.add_resource(TransQueue::<T, StateEvent>::with_capacity(2));                                                                           
    |               ^^^^^^^^^^^^                                                                                                                           
                                                                                                                                                           
error: aborting due to 2 previous errors                                                                                                                   
                                                                                                                                                           
For more information about this error, try `rustc --explain E0310`.                                                                                        
error: Could not compile `amethyst`.
```

In my code I have T: 'a